### PR TITLE
docs: add langchain integration docs

### PR DESCRIPTION
**Pull request by e-moran**
The package.json needs to be updated in these docs once the langchain integration is released, will leave it in draft for now because its not merge-able yet.

### CODE REVIEW

**Review by davidmytton — COMMENTED** [review]
Fixed some typos. When https://github.com/langchain-ai/langchainjs/pull/6674 is merged we can link to their docs here as well, then we can merge this.

**Comment by davidmytton** [comment]
@SocketSecurity ignore npm/commander@10.0.1
@SocketSecurity ignore npm/langsmith@0.1.55

File: src/content/docs/integrations/snippets/LangChainChatModel.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  contextWindowSize: 1,
+
+  // This allows you to provide custom replacements when redacting. Please ensure
+  // that the replacements are unique so that unredaction works as expected.
+  replace: (identifiedType: string) => {
+    return identifiedType === "email" ? "redacted@example.com" : undefined;
+  },
+};
+
+const arcjetRedact = new ArcjetRedact(arcjetRedactOptions);
```
Comment (davidmytton) [inline src/content/docs/integrations/snippets/LangChainChatModel.ts:38]: @blaine-arcjet could you have a look at this for me please? There's a TS error:

```
Argument of type '{ chatModel: ChatOpenAI<ChatOpenAICallOptions>; entities: string[]; detect: (tokens: string[]) => ("custom-entity" | undefined)[]; contextWindowSize: number; replace: (identifiedType: string) => "redacted@example.com" | undefined; }' is not assignable to parameter of type 'ArcjetRedactOptions<(tokens: string[]) => ("custom-entity" | undefined)[]>'.
  Types of property 'entities' are incompatible.
    Type 'string[]' is not assignable to type '(ArcjetSensitiveInfoType | "custom-entity")[]'.ts(2345)
```

Comment (blaine-arcjet) [reply src/content/docs/integrations/snippets/LangChainChatModel.ts:38]: It seems that this type inference doesn't work if you don't pass the options object directly. I can push something up shortly.

Comment (blaine-arcjet) [reply src/content/docs/integrations/snippets/LangChainChatModel.ts]: @davidmytton this is building now.

**Pull request merged**